package/lib/src/beam_guard.dart

```diff
@@ -91,8 +93,18 @@ class BeamGuard {
             return true;
           }
         } else {
-          if (pathBlueprint == location.state.uri.toString()) {
-            return true;
+          if (location.state.uri.hasQuery) {
```

Comment:
I think you could go with just `.path` instead of this manual logic;
https://api.dart.dev/stable/2.13.1/dart-core/Uri/path.html

Reply:
Good point! I changed it :)